$NetBSD: patch-Makefile,v 1.2 2022/08/30 17:55:31 schmonz Exp $

Do not consider building a local copy of BearSSL.
Apply upstream commit 0cb7bb4 to fall back to /dev/urandom on systems
where getentropy() is not present.

--- Makefile.orig	2022-08-14 12:18:16.000000000 +0000
+++ Makefile
@@ -10,6 +10,13 @@ BINARIES+=tlswrapper-test
 
 all: bearssl $(BINARIES) tlswrapper-tcp tlswrapper-smtp
 
+randombytes.h:
+	(grep -v "randombytes.h" "randombytes.c-01getentropy"; echo "int main() {}";) > try.c
+	[ ! -f randombytes.h ] && $(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) -o try try.c && cat randombytes.h-01getentropy > randombytes.h || :
+	(grep -v "randombytes.h" "randombytes.c-02devurandom"; echo "int main() {}";) > try.c
+	[ ! -f randombytes.h ] && $(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) -o try try.c && cat randombytes.h-02devurandom > randombytes.h || :
+	rm try.c try
+
 alloc.o: alloc.c log.h alloc.h
 	$(CC) $(CFLAGS) $(CPPFLAGS) -c alloc.c
 
@@ -267,9 +274,6 @@ tlswrapper-test: tlswrapper-test.o $(OBJ
 
 
 bearssl:
-	echo 'int main(){}' > try.c
-	$(CC) $(CFLAGS) $(CPPFLAGS) -o try.o $(LDFLAGS) try.c || (sh bearssl.sh; cd bearssl; make; rm build/*.so; )
-	rm -f try.o try.c
 	mkdir -p bearssl/inc
 
 tlswrapper-tcp: tlswrapper
@@ -296,4 +300,5 @@ test: bearssl $(BINARIES) tlswrapper-tcp
 
 clean:
 	rm -f *.o *.out $(BINARIES) tlswrapper-tcp tlswrapper-smtp
+	rm -f randombytes.h
 
